### DISCUSSION TITLE
DataTypeDefinition Attribute

### DISCUSSION THREAD

**Discussion by V-F** [eclipse-milo/milo#798] [Ideas]
Hello @kevinherron,

is it planned that by creation of custom DataTypes the [`DataTypeDefinition`](https://reference.opcfoundation.org/v104/Core/docs/Part3/8.48/) attribute will be added to the [`DataType Node`](https://reference.opcfoundation.org/v104/Core/docs/Part3/5.8.3/)?
If not what is the simplest way to add this attribute yourself during the creation of namespace (if it is possible)?
I mean the API access to the `UaDataTypeNode` for calling `setProperty()`-method (since in the [constant for DataTypeDefinition-attribute](https://reference.opcfoundation.org/v104/Core/docs/Part6/A.1/) isn't defined in the  ​AttributeId-class). Then I can use the `getPropertyNodeAsync()`-method on the client side for reading it. Please correct me if I'm wrong with usage of methods.

It would be nice if the DataTypeDictionary (or even DefaultDataTypeManager) will have some API for getting `TypeDescription` for custom DataTypes (currently I use the modified version of `DataTypeDictionaryReader` in the modifed version of `DataTypeDictionarySessionInitializer` for collecting this info since I need the info about [`FieldType`](https://reference.opcfoundation.org/v104/Core/docs/Part5/E.2.6/)'s of the custom `StructuredType`'s).

Thanks in advance for your answer

**Comment by kevinherron** [comment]
That is part of OPC UA 1.04 which is not currently implemented. I "stole" these handful of built-in structs related to defining structures from 1.04 but otherwise Milo currently implements 1.03, so there is no definition attribute on DataType Node at the moment (it is defined as an attribute by OPC UA 1.04, not a separate Property Node).

I think what you are asking will not technically be possible until I start implementing OPC UA 1.04 in version 2.0 of the library (work will begin soon).

**Reply by V-F** [reply]
Thank you for your quick and comprehensive answer.
